plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CallMediatorTransformer.java

```diff
@@ -117,17 +120,26 @@ public void transformWithinSequence(TransformationInfo information, EsbNode subj
         return synapseCallMediator;
     }
 
-    public ResolvingEndpoint getEndpointFromXpath(NamespacedProperty nameSpacedProperty) throws JaxenException {
-
-        SynapseXPath synapseXPath = new SynapseXPath(nameSpacedProperty.getPropertyValue());
-        for (int i = 0; i < nameSpacedProperty.getNamespaces().keySet().size(); ++i) {
-            String prefix = (String) nameSpacedProperty.getNamespaces().keySet().toArray()[i];
-            String namespaceUri = nameSpacedProperty.getNamespaces().get(prefix);
-            synapseXPath.addNamespace(prefix, namespaceUri);
+    public static ResolvingEndpoint getEndpointFromXpath(NamespacedProperty nameSpacedProperty, boolean isForValidation)
+            throws JaxenException {
+        SynapseXPath synapseXPath;
+        ResolvingEndpoint resolvingEndpoint;
+        if (StringUtils.isEmpty(nameSpacedProperty.getPropertyValue()) && !isForValidation) {
+            synapseXPath = new SynapseXPath("FACTORY/DEFAULT/VALUE");
```

Comment:
Let's move this to a meaningful constant visible across all mediator implementations.

Reply:
Fixed